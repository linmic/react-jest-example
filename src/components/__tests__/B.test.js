import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<B />', () => {
  let B;
  let makeShallowSubject;
  let makeSubject;

  beforeEach(() => {
    B = require('../B').default;

    makeShallowSubject = () => shallow(<B />);
    makeSubject = () => mount(<B />);
  });

  it('should match shallow snapshot', () => {
    const subject = makeShallowSubject();

    expect(toJson(subject)).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    const subject = makeSubject();

    expect(toJson(subject)).toMatchSnapshot();
  });
});
