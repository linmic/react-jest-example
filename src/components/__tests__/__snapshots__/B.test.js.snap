// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<B /> should match shallow snapshot 1`] = `
<nav>
  <C
    id={0}
    key="0"
  />
  <C
    id={1}
    key="1"
  />
  <C
    id={2}
    key="2"
  />
</nav>
`;

exports[`<B /> should match snapshot 1`] = `
<B>
  <nav>
    <C
      id={0}
      key="0"
    >
      <a
        href="#"
      >
        I am link 
        0
        .
      </a>
    </C>
    <C
      id={1}
      key="1"
    >
      <a
        href="#"
      >
        I am link 
        1
        .
      </a>
    </C>
    <C
      id={2}
      key="2"
    >
      <a
        href="#"
      >
        I am link 
        2
        .
      </a>
    </C>
  </nav>
</B>
`;
